qiskit_nature/problems/second_quantization/molecular/fermionic_op_builder.py

```diff
@@ -0,0 +1,128 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Fermionic operator builder. """
+
+import itertools
+from typing import List, Tuple
+
+import numpy as np
+
+from qiskit_nature.drivers.qmolecule import QMolecule
+from qiskit_nature.operators import FermionicOp
+
+
+def build_fermionic_op(q_molecule: QMolecule) -> FermionicOp:
+    """
+    Builds a fermionic operator based on a QMolecule object.
+
+    Args:
+        q_molecule (QMolecule): QMolecule instance with 1- and/or 2-body integrals.
+
+    Returns:
+        FermionicOp: FermionicOp built from a QMolecule object.
+    """
+
+    one_body_ints = q_molecule.one_body_integrals
+    two_body_ints = q_molecule.two_body_integrals
+
+    return build_ferm_op_from_ints(one_body_ints, two_body_ints)
+
+
+def build_ferm_op_from_ints(one_body_integrals: np.ndarray,
+                            two_body_integrals: np.ndarray = None) -> FermionicOp:
+    """
+    Builds a fermionic operator based on 1- and/or 2-body integrals. Integral values are used for
+    the coefficients of the second-quantized Hamiltonian that is built. If integrals are stored
+    in the '*chemist*' notation
+             h2(i,j,k,l) --> adag_i adag_k a_l a_j
+    they are required to be in block spin format and also have indexes reordered as follows
+    'ijkl->ljik'.
+    There is another popular notation, the '*physicist*' notation
+             h2(i,j,k,l) --> adag_i adag_j a_k a_l
+    If you are using the '*physicist*' notation, you need to convert it to
+    the '*chemist*' notation. E.g. h2=numpy.einsum('ikmj->ijkm', h2)
+    The :class:`~qiskit_nature.QMolecule` class has
+    :attr:`~qiskit_nature.QMolecule.one_body_integrals` and
+    :attr:`~qiskit_nature.QMolecule.two_body_integrals` properties that can be
+    directly supplied to the `h1` and `h2` parameters here respectively.
+
+    Args:
+        one_body_integrals (numpy.ndarray): One-body integrals stored in the chemist notation.
+        two_body_integrals (numpy.ndarray): Two-body integrals stored in the chemist notation.
+
+    Returns:
+        FermionicOp: FermionicOp built from 1- and/or 2-body integrals.
+    """
+
+    fermionic_op = _build_fermionic_op(one_body_integrals, two_body_integrals)
+    fermionic_op = fermionic_op.reduce()
+
+    return fermionic_op
+
+
+def _build_fermionic_op(one_body_integrals: np.ndarray,
+                        two_body_integrals: np.ndarray) -> FermionicOp:
+    one_body_base_ops_labels = _create_one_body_base_ops(one_body_integrals)
+    two_body_base_ops_labels = _create_two_body_base_ops(
+        two_body_integrals) if two_body_integrals is not None else []
+    base_ops_labels = one_body_base_ops_labels + two_body_base_ops_labels
+    initial_label_with_ceoff = ('I' * len(one_body_integrals), 1)
```

Comment:
I just did some calculations locally which are sort-of related to this and I noticed that we should ensure that the coefficient here is actually 0 rather than 1:
```suggestion
    initial_label_with_ceoff = ('I' * len(one_body_integrals), 0)
```

Reply:
My bad! I looked at this piece of code from the `FermionicOp` constructor in the context of the original code snippet that passed the dummy label in the constructor.
```
        elif isinstance(data, str):
            label = data
            if not self._validate_label(label):
                raise QiskitNatureError(
                    "Label must be a string consisting only of "
                    f"['I','+','-','N','E'] not: {label}"
                )
            self._register_length = len(label)
            self._labels = [label]
            self._coeffs = [1]
```

Reply:
Yes no worries at all! I think I also made that mistake in my original snippet but I noticed yesterday that this leads to an energy offset of `1.0` further down in the stack.

Reply:
But it's a good point that you bring up that snippet! We should discuss whether to change the default coeff to `0`. Let's do that offline and do that in another PR which is likely to happen on the `FermionicOp` anyways.

Reply:
This will be addressed in #68 :+1: